### PR TITLE
Profile plot refac

### PR DESCRIPTION
**Pull request by agenius-logesh**
Removed parameters.
Colors matched according to the group levels.
gynSymCol changed to gynNameCol.
foldChangeLines changed to foldChangeThreshold and log2() is entered inside the code.
DGEdata input changed to DGEobj.
For ggplot no gynSymLables - removed.

### CODE REVIEW

File: R/profilePlot.R [hunk 1]
```diff
@@ -134,20 +123,28 @@ profilePlot <- function(contrastDF,
...
-    if (!missing(geneSymCol)) {
-        assertthat::assert_that(!is.null(geneSymCol),
-                                geneSymCol %in% colnames(contrastDF),
-                                msg = "geneSymCol column not found in contrastDF.")
+                            msg = "pvalCol to be a singular value of class character and must be in contrast data.")
+
+    if (!missing(geneNameCol)) {
+        assertthat::assert_that(!is.null(geneNameCol),
+                                length(geneNameCol) == 1,
+                                geneNameCol %in% names(DGEobj::getType(dgeObj, type = "geneData")[[1]]),
```
Comment (agenius-mohammed-ali) [inline R/profilePlot.R:159]: - There must be check for the existence of geneData in the passed DGEobj
- There must be check for the geneData length

File: R/profilePlot.R [hunk 2]
```diff
@@ -337,28 +273,29 @@ profilePlot <- function(contrastDF,
...
 
-        if (!missing(geneSymCol)) {
-            var.annot <- merge(var.annot, contrastDF %>%
-                                   dplyr::select(!!dplyr::sym(geneSymCol)),
-                               by = 0)
-            rownames(var.annot) <- var.annot$Row.names
-            var.annot$Row.names <- NULL
+        if (!missing(geneNameCol)) {
+            gene_data <- DGEobj::getItem(dgeObj, "geneData") %>%
+                dplyr::select(all_of(geneNameCol))
```
Comment (agenius-mohammed-ali) [inline R/profilePlot.R]: - It is one column so why using `all_of` function.
- missing `dplyr::` before `all_of`

File: R/profilePlot.R [hunk 3]
```diff
@@ -246,36 +195,13 @@ profilePlot <- function(contrastDF,
...
     if (any(is.null(sizeBySignificance),
             !is.logical(sizeBySignificance),
             length(sizeBySignificance) != 1)) {
         warning("sizeBySignificance must be a singular logical value. Assigning default value FALSE")
         sizeBySignificance = FALSE
     }
+
     # Columns to plot
     # Capture the labels from the colname
     xlabel <- make.names(logIntCol)
```
Comment (agenius-mohammed-ali) [inline R/profilePlot.R:220]: why not using make names on all column names one time?

**Review by agenius-mohammed-ali — CHANGES_REQUESTED** [review]

File: R/profilePlot.R (hunk 3 above)
Comment (agenius-logesh) [reply R/profilePlot.R:220]: this is did by you only so I cant get your point mohammed.

File: R/profilePlot.R (hunk 2 above)
Comment (agenius-logesh) [reply R/profilePlot.R]: It makes a waring message if we not using all_of()

File: R/profilePlot.R (hunk 1 above)
Comment (agenius-logesh) [reply R/profilePlot.R:159]: done.

**Pull request closed without merging**